src/tcp.c

```diff
@@ -458,7 +458,7 @@ tcp_socket_dead(int fd)
   if (err < 0)
     return -errno;
   else if (err == 0)
-      return -EIO;
```

Comment:
It looks like that 3895c923a3a959da05080831b8146c09ed143b00 should be reverted ?